src/Core/ServerSettings.h

```diff
@@ -96,7 +96,8 @@ namespace DB
     M(UInt64, total_memory_profiler_step, 0, "Whenever server memory usage becomes larger than every next step in number of bytes the memory profiler will collect the allocating stack trace. Zero means disabled memory profiler. Values lower than a few megabytes will slow down server.", 0) \
     M(Double, total_memory_tracker_sample_probability, 0, "Collect random allocations and deallocations and write them into system.trace_log with 'MemorySample' trace_type. The probability is for every alloc/free regardless to the size of the allocation (can be changed with `memory_profiler_sample_min_allocation_size` and `memory_profiler_sample_max_allocation_size`). Note that sampling happens only when the amount of untracked memory exceeds 'max_untracked_memory'. You may want to set 'max_untracked_memory' to 0 for extra fine grained sampling.", 0) \
     M(UInt64, total_memory_profiler_sample_min_allocation_size, 0, "Collect random allocations of size greater or equal than specified value with probability equal to `total_memory_profiler_sample_probability`. 0 means disabled. You may want to set 'max_untracked_memory' to 0 to make this threshold to work as expected.", 0) \
-    M(UInt64, total_memory_profiler_sample_max_allocation_size, 0, "Collect random allocations of size less or equal than specified value with probability equal to `total_memory_profiler_sample_probability`. 0 means disabled. You may want to set 'max_untracked_memory' to 0 to make this threshold to work as expected.", 0)
+    M(UInt64, total_memory_profiler_sample_max_allocation_size, 0, "Collect random allocations of size less or equal than specified value with probability equal to `total_memory_profiler_sample_probability`. 0 means disabled. You may want to set 'max_untracked_memory' to 0 to make this threshold to work as expected.", 0) \
+    M(Bool, validate_tcp_client_information, true, "Validate client_information in the query packet over the native TCP protocol. ClickHouse-client's with versions 23.7 can't send queries to newer servers with `validate_tcp_client_information` enabled.", 0)
```

Comment:
NOTE:  
This option is enabled by default. I'm not absolutely sure that it is better than the disabled option. But I would prefer maximum security enabled by default.

The upgrade process should include simultaneous upgrades of client and server, like this:

```bash
$ sudo apt-get update
$ sudo apt-get install clickhouse-client clickhouse-server
$ sudo service clickhouse-server restart
```

On all ClickHouse hosts.

--- --- --- 
with disabled  validate_tcp_client_information by default option we also need to make tests  `02865_tcp_proxy_query_packet_validation` pass. 
To do that:
Create an XML tests/config/config.d/validate_tcp_client_information.xml file:
```xml
<clickhouse>
    <validate_tcp_client_information>true</validate_tcp_client_information>
</clickhouse>
```

And include this file in tests/config/install.sh file:
```bash
ln -sf $SRC_PATH/config.d/validate_tcp_client_information.xml $DEST_SERVER_PATH/config.d/
```

Reply:
Update:
It was decided to disable client information validation by default because there are many libraries that already use the native protocol: https://clickhouse.com/docs/en/interfaces/third-party/client-libraries

Reply:
I think we will enable it by default in the next LTS (24.3)

Reply:
Yes, it is false for now